src/borrower/nftfeed/wrapper.sol

```diff
@@ -0,0 +1,34 @@
+// principal.sol
+// Copyright (C) 2020 Centrifuge
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.5.3;
+
+import "./nftfeed.sol";
+
+contract NFTFeedWrapper is NFTFeed {
+    // Ceiling Wrapper
+
+    function borrow(uint loan, uint amount) external auth {
+        // ceiling check uses existing loan debt
+        require(ceiling(loan) >= safeAdd(pile.debt(loan), amount), "borrow-amount-too-high");
+    }
+
+    function repay(uint loan, uint amount) external auth {}
+
+    // Threshold Wrapper
+    // todo
```

Comment:
Since the collector threshold is already a function call to an external contract. I won't implement a wrapper and modify the collector contract instead